tests/Functional/Parameters/BooleanFilterTest.php

```diff
@@ -0,0 +1,187 @@
+
+    /**
+     * @return class-string[]
+     */
+    public static function getResources(): array
+    {
+        return [FilteredBooleanParameter::class];
+    }
+
+    /**
+     * @throws DecodingExceptionInterface
+     * @throws ClientExceptionInterface
+     * @throws ServerExceptionInterface
+     * @throws RedirectionExceptionInterface
+     * @throws TransportExceptionInterface
+     */
+    public function testBooleanFilterWithValidValue(): void
+    {
+        $resource = $this->isMongoDB() ? FilteredBooleanParameterDocument::class : FilteredBooleanParameter::class;
+        $this->recreateSchema([$resource]);
+        $this->loadFixtures($resource);
+
+        $this->assertBooleanFilterResponse('true', 2);
+        $this->assertBooleanFilterResponse('false', 1);
+        $this->assertBooleanFilterResponse('1', 2);
+        $this->assertBooleanFilterResponse('0', 1);
+    }
+
+    /**
+     * @throws RedirectionExceptionInterface
+     * @throws DecodingExceptionInterface
+     * @throws ClientExceptionInterface
+     * @throws TransportExceptionInterface
+     * @throws ServerExceptionInterface
+     */
+    public function testBooleanFilterWithInvalidValue(): void
+    {
+        $this->assertBooleanFilterResponse('null', 3); // <=> ignored
+        $this->assertValidationErrorResponse('');
+    }
+
+    /**
+     * @throws DecodingExceptionInterface
+     * @throws ClientExceptionInterface
+     * @throws ServerExceptionInterface
+     * @throws RedirectionExceptionInterface
+     * @throws TransportExceptionInterface
+     */
+    public function testBooleanFilterAliasWithValidValue(): void
+    {
+        $resource = $this->isMongoDB() ? FilteredBooleanParameterDocument::class : FilteredBooleanParameter::class;
+        $this->recreateSchema([$resource]);
+        $this->loadFixtures($resource);
+
+        $this->assertBooleanFilterResponse('true', 2, 'enabled');
+        $this->assertBooleanFilterResponse('false', 1, 'enabled');
+        $this->assertBooleanFilterResponse('1', 2, 'enabled');
+        $this->assertBooleanFilterResponse('0', 1, 'enabled');
+    }
+
+    /**
+     * @throws RedirectionExceptionInterface
+     * @throws DecodingExceptionInterface
+     * @throws ClientExceptionInterface
+     * @throws TransportExceptionInterface
+     * @throws ServerExceptionInterface
+     */
+    public function testBooleanFilterAliasWithInvalidValue(): void
+    {
+        $this->assertBooleanFilterResponse('null', 3, 'enabled'); // <=> ignored
+        $this->assertValidationErrorResponse('', 'enabled');
+    }
+
+    /**
+     * @throws TransportExceptionInterface
+     * @throws ServerExceptionInterface
+     * @throws RedirectionExceptionInterface
+     * @throws DecodingExceptionInterface
+     * @throws ClientExceptionInterface
+     */
+    private function assertBooleanFilterResponse(?string $activeValue, int $expectedCount, string $param = 'active'): void
+    {
+        $route = $this->getRoute($activeValue, $param);
```

Comment:
let's avoid these kind of functions, you should rather use a phpunit data provider and use different values inside the test.

Reply:
Refactored using a DataProvider with a Generator.